src/dom.ts

```diff
@@ -158,3 +179,44 @@ export function isVisible (element: Element) {
     /* } */
     /* return true */
 }
+
+/** Get all elements that match the given selector
+ *
+ * @param selector   `the CSS selector to choose elements with
+ * @param filter      filter to use to further chose items, or null for all
+ */
+export function getElemsBySelector(selector: string, filter: ElementFilter) {
+    let elems = Array.from(document.querySelectorAll(selector))
+
+    return filter ? elems.filter(filter) : elems
+}
+
+/** Get the nth input element on a page
+ *
+ * @param nth         the element index, can be negative to start at the end
+ * @param filter      filter to use to further chose items, or null for all
+ */
+export function getNthElement(selectors: string, nth: number,
+    filter: ElementFilter) {
+
+    let inputs = getElemsBySelector(selectors, filter)
+
+    if (inputs.length) {
+        let index = Number(nth).clamp(-inputs.length, inputs.length - 1)
+            .mod(inputs.length)
+
+        return <HTMLElement>inputs[index]
+    }
+
+    return null
+}
+
+/** Sorter functon by offsetWidth/Height, used for sorting elements by their
+ *  area on the page
+ */
+export function elementAreaSorter(a: HTMLElement, b: HTMLElement): number {
+    const aArea = a.offsetWidth * a.offsetHeight
+    const bArea = b.offsetWidth * b.offsetHeight
+
+    return (aArea > bArea) ? 1 : -1
```

Comment:
compareFunction is supposed to return 0 if the elements are equal.

Suggest renaming to compareElementArea (because it doesn't actually sort) and getting the equal condition right with: `return aArea - bArea`